sdk-core/src/main/java/care/data4life/sdk/network/NetworkingContract.kt

```diff
@@ -84,23 +86,44 @@ interface NetworkingContract {
         fun fetchVersionInfo(): Single<VersionList>
     }
 
-    interface Interceptor : okhttp3.Interceptor {
+    internal interface CertificatePinnerFactory {
+        fun getInstance(platform: String, env: Environment): CertificatePinner
+    }
+
+    internal interface Interceptor : okhttp3.Interceptor {
         override fun intercept(chain: okhttp3.Interceptor.Chain): Response
     }
 
-    interface CertificatePinnerFactory {
-        fun getInstance(baseUrl: String, pin: String): CertificatePinner
+    internal interface PartialInterceptor<T : Any> {
```

Comment:
More of a comment/question: 

Is there a need for actually creating the concept of a `PartialInterceptor`? 
Might it be worth having only plain interceptors to keep the interfaces simple? In my understanding, it feels more like a namespacing/dependency issue than anything else (as in retryToken and oauthFetchToken both are needed and make sense in the context of Oauth interceptor), but the cost of having a partial interface added a lot overhead on understanding whats happening.

Moreover when does an interceptor become partial? technically all of them could be partial

Reply:
The Problem which it solves is, that interceptors are basically visitors. We have currently the case with AccessToken, that it does something before the chain proceeds and after the the request returns. And this actions are depend. Since no order of execution is guaranteed, I needed to clue them together in way, that they are still testable.
I do not like it either, but I had no better idea so far...
However if you have...feel free to let me know.

And this belongs to the Interceptor PR.

Reply:
it's a good solution already. I was just guessing if it was possible to unify them somehow, but i cant think of how right now.